src/kiln_plugin.h

```diff
@@ -0,0 +1,255 @@
+    LRCQW_RECEIVE_AS_TOKENS__ITEMS,
+
+    LRCQW_UNEXPECTED_PARAMETER
+
+} lr_complete_queued_withdrawals_parameters;
+
+typedef enum {
+    LR_DELEGATE_TO_OPERATOR = 0,
+    LR_DELEGATE_TO_SIGNATURE_OFFSET,
+    LR_DELEGATE_TO_APPROVER_SALT,
+    LR_DELEGATE_TO_SIGNATURE_SIG_OFFSET,
+    LR_DELEGATE_TO_SIGNATURE_EXPIRY,
+    LR_DELEGATE_TO_SIGNATURE_SIG_LENGTH,
+    LR_DELEGATE_TO_SIGNATURE_SIG_ITEMS,
+    LR_DELEGATE_TO_UNEXPECTED_PARAMETER
+} lr_delegate_to_parameters;
+
+// ****************************************************************************
+
+// Parsing structures
+
+typedef struct {
+    int strategy_to_display;
+    int erc20_to_display;
+    uint8_t erc20_amount_to_display[INT256_LENGTH];
+} lr_deposit_t;
+
+typedef struct {
+    // -- utils
+    uint16_t current_item_count;
+    // -- display
+    char operator_address[ADDRESS_STR_LEN];
+    bool is_kiln;
+} lr_delegate_to_t;
+
+typedef struct {
+    //  -- utils
+    uint16_t queued_withdrawals_count;
+    uint16_t current_item_count;
+
+    // -- display
+    uint8_t strategies_count;
+    char withdrawer[ADDRESS_STR_LEN];
+    // list of strategies indexes **INCREMENTED BY 1** to display in the UI
+    // 0 is reserved for end of array
+    // UNKNOW_LR_STRATEGY is used to display the "unknown" strategy
+    // assumptions:
+    // (i) in practice, we should not encounter more than
+    //      LR_STRATEGIES_COUNT +~ a few potential unsupported
+    //      strategies in the plugin. So * 3 should be a good enough buffer.
+    // (ii) in practice there should not be more than (2 ** 8) - 2 known strategies
+    uint8_t strategies[MAX_DISPLAYABLE_LR_STRATEGIES_COUNT];
+} lr_queue_withdrawals_t;
+
+typedef struct {
+    // -- utils
+    uint16_t parent_item_count;
+    uint16_t current_item_count;
+    // -- total values
+    uint16_t relegations_count;
+    uint8_t withdrawals_count;
+    uint16_t strategies_count;
+
+    // -- display
+    char withdrawer[ADDRESS_STR_LEN];
+    // list of strategies indexes **INCREMENTED BY 1** to display in the UI
+    // 0 is reserved for end of array
+    // UNKNOW_LR_STRATEGY is used to display the "unknown" strategy
+    // assumptions:
+    // (i) in practice, we should not encounter more than
+    //      LR_STRATEGIES_COUNT +~ a few potential unsupported
+    //      strategies in the plugin. So * 3 should be a good enough buffer.
+    // (ii) in practice there should not be more than (2 ** 8) - 2 known strategies
+    uint8_t strategies[MAX_DISPLAYABLE_LR_STRATEGIES_COUNT];
+    // follows the indexes of the strategies array in this structure.
+    // value is the withdrawal number
+    // assumptions:
+    // (i) in practice, we should not encounter more than
+    //      255 strategies in queued withdrawal calls
+    uint8_t withdrawals[MAX_DISPLAYABLE_LR_STRATEGIES_COUNT];
+    // follows the indexes of the strategies array in this structure
+    bool is_redelegated[MAX_DISPLAYABLE_LR_STRATEGIES_COUNT];
```

Comment:
I think we should change the name here. the param in the contract is called `receiveAsTokens`. if set to `true` it will send you the native ETH / LST of the withdrawal, else it will re-inject the shares in the corresponding strategies